src/ol/geom/CircularString.js

```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * @private
+     * @type {number}
+     */
+    this.maxDelta_ = -1;
+
+    /**
+     * @private
+     * @type {number}
+     */
+    this.maxDeltaRevision_ = -1;
+
+    if (opt_layout !== undefined && !Array.isArray(coordinates[0])) {
+      this.setFlatCoordinates(
+        opt_layout,
+        /** @type {Array<number>} */ (coordinates)
+      );
+    } else {
+      this.setCoordinates(
+        /** @type {Array<import("../coordinate.js").Coordinate>} */ (
+          coordinates
+        ),
+        opt_layout
+      );
+    }
+  }
+
+  /**
+   * Make a complete copy of the geometry.
+   * @return {!CircularString} Clone.
+   * @api
+   */
+  clone() {
+    const circularString = new CircularString(
+      this.flatCoordinates.slice(),
+      this.layout
+    );
+    circularString.applyProperties(this);
+    return circularString;
+  }
+
+  /**
+   * @param {number} x X.
+   * @param {number} y Y.
+   * @param {import("../coordinate.js").Coordinate} closestPoint Closest point.
+   * @param {number} minSquaredDistance Minimum squared distance.
+   * @return {number} Minimum squared distance.
+   */
+  closestPointXY(x, y, closestPoint, minSquaredDistance) {
+    if (minSquaredDistance < closestSquaredDistanceXY(this.getExtent(), x, y)) {
+      return minSquaredDistance;
+    }
+    if (this.maxDeltaRevision_ !== this.getRevision()) {
+      this.maxDelta_ = Math.sqrt(
+        maxSquaredDelta(
+          this.flatCoordinates,
+          0,
+          this.flatCoordinates.length,
+          this.stride,
+          0
+        )
+      );
+      this.maxDeltaRevision_ = this.getRevision();
+    }
+    return assignClosestPoint(
+      this.flatCoordinates,
+      0,
+      this.flatCoordinates.length,
+      this.stride,
+      this.maxDelta_,
+      false,
+      x,
+      y,
+      closestPoint,
+      minSquaredDistance
+    );
+  }
+
+  /**
+   * Return the coordinates of the circular string.
+   * @return {Array<import("../coordinate.js").Coordinate>} Coordinates.
+   * @api
+   */
+  getCoordinates() {
+    return inflateCoordinates(
+      this.flatCoordinates,
+      0,
+      this.flatCoordinates.length,
+      this.stride
+    );
+  }
+
+  /**
+   * Get the type of this geometry.
+   * @return {import("./GeometryType.js").default} Geometry type.
+   * @api
+   */
+  getType() {
+    return GeometryType.CIRCULAR_STRING;
+  }
+
+  /**
+   * Test if the geometry and the passed extent intersect.
+   * @param {import("../extent.js").Extent} extent Extent.
+   * @return {boolean} `true` if the geometry and the extent intersect.
+   * @api
+   */
+  intersectsExtent(extent) {
+    // test per arc extents
+    for (let i = 0; i < this.flatCoordinates.length - 2; i += 4) {
+      const arcExtents = extendFlatCoordinates(
+        createEmpty(),
+        this.flatCoordinates,
+        i,
+        i + this.stride * 3,
+        this.stride
+      );
+      if (intersects(extent, arcExtents)) {
+        return true;
+      }
+    }
```

Comment:
Deze code snap ik niet goed, het lijkt erop dat telkens de bounding box om het beginpunt van het boogstuk en het eindpunt van het boogstuk wordt berekend. En dat is niet perse gelijk aan de boundingbox die de werkelijke boog beslaat.

Ik dacht dat dit de reden was dat een boog soms niet getekend wordt, maar ik krijg het niet gedebugged.

*Voorbeeld 1*
`01080000000300000000000000000008400000000000001c40000000000000f03f0000000000001c4000000000000000000000000000001840`

Als ik hier aan de rechterkant inzoom dan kan ik maximaal gaan en wordt de boog getekend, zelfs als deze bijna recht is. Aan de linkerkant inzoomen geeft bij mij al snel dat de boog niet getekend wordt. Mogelijk ligt het aan de positie van de boogpunten, maar ik kan er geen regelmaat in vinden.

*Voorbeeld 2*
`0108000000030000000000000000001040000000000000204000000000000014400000000000002240fea1ad44160a0a40117abfd9f7d91d40`

Deze bijna-cirkel wordt gemaakt door boogpunten die redelijk dicht bij elkaar liggen. Als ik hiermee de boog langzaam naar beneden schuif (pannen) dan wordt deze al snel niet getekend, terwijl de boog eigenlijk nog vol in beeld is.


Het zal iets met de extents en boogpunten te maken hebben, misschien in een andere functie?

Reply:
> Deze code snap ik niet goed, het lijkt erop dat telkens de bounding box om het beginpunt van het boogstuk en het eindpunt van het boogstuk wordt berekend. En dat is niet perse gelijk aan de boundingbox die de werkelijke boog beslaat.

De intentie was een boundingbox om begin-, eind- en middenpunt. Ik heb geredeneerd dat je dan toch altijd een box hebt waarin de hele boog zit? Of eigenlijk meerdere boxen in dit geval ...

Ik weet nu even niet meer of dat ook gelukt is. Maar ook ik kreeg het dus niet gedebugged.

Reply:
Goed, net iets langer geredeneerd en dat middenpunt ligt natuurlijk niet altijd in het midden van de boog. Ik denk dat ik daar al mis ging. Dat gezegd hebbende ben je er dan nog niet, afhankelijk van de grote van de boog moet ik mogelijk nog meer punten bepalen. Ik ga dit verbeteren, en kijken of ik het op één of andere manier kan debuggen. Het moet hoe dan ook kloppen denk ik.

Reply:
![image](https://user-images.githubusercontent.com/65287900/147370760-a996088b-32cd-46a6-866b-9c7a12934bfd.png)

Ik heb het bepalen van de bounding boxes denk ik ook op orde, na de kerst zal ik het toevoegen :smile: